File: python/arcade-intro/03_Smooth Sailing/009_allLongestStrings.py
def allLongestStrings(inputArray):
    '''
    Given an array of strings, return another array containing all of its longest strings.
    '''
    resultArray = []
    currentMax = len(inputArray[0])
    
    for i in range(len(inputArray) ):
        
        currentLen = len(inputArray[i])        
        if currentLen > currentMax :
            resultArray = [inputArray[i]]
            currentMax = currentLen
        elif currentLen == currentMax :
            resultArray.append(inputArray[i])
            
    return resultArray

print(allLongestStrings(["aba", "aa", "ad", "vcd", "aba"]))

File: arcade/go/arcade-intro/03_Smooth Sailing/009_allLongestStrings.go
func allLongestStrings(inputArray []string) []string {
	var result []string

	if len(inputArray) == 0 {
		return result
	}

	maxLength := -1
	for _, s := range inputArray {
		if len(s) == maxLength {
			result = append(result, s)
		} else if len(s) > maxLength {
			result = []string{s}
			maxLength = len(s)
		}
	}

	return result
}
